### DISCUSSION TITLE
collection has not been loaded to memory or load failed

### DISCUSSION THREAD

**Discussion by dkrakau** [milvus-io/milvus#22944] [Q&A and General discussion] [answered]
Hey everyone,

I have implemented the milvus "book" tutorial using java and milvus 2.2.3 (docker).
Inserting new entries works fine, but when i want to search for a vector i get this Error:

collection <collection_id> has not been loaded to memory or load failed

Does somebody know why this error occures?

**Comment by NehaTamore** [comment]
Could you share more specifics about the error? 
Have you tried the following? 
from pymilvus import Collection
collection = Collection("book")      # Get an existing collection.
**collection.load(replica_number=2)**

Also, when we load the collection, we must ensure adequate memory is available.

**Comment by yhmo** [comment, marked as answer]
NehaTamore is right. A collection needs to be loaded into memory before searching.
In Milvus, to search for a collection, two pre-conditions are required:
1. create an index for the collection's vector field
2. load the collection into memory

**Comment by dkrakau** [comment]
Oh Thank you very much, i forgot to create an index for my vector field first.
